File: Digby/class_10/SplitVewController/SplitVewController/TableViewController.m
//
//  TableViewController.m
//  SplitVewController
//
//  Created by Digby Andrews on 2016-07-04.
//  Copyright © 2016 Digby Andrews. All rights reserved.
//

#import "TableViewController.h"

@interface TableViewController () <UITableViewDelegate, UITableViewDataSource>

@property (weak, nonatomic) IBOutlet UITableView *tableView;

@end

@implementation TableViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    self.tableView.delegate = self;
    self.tableView.dataSource = self;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    UITableViewCell *cell = [self.tableView dequeueReusableCellWithIdentifier:@"CellIdentifier"];
    if (!cell)
    {
        cell = [[UITableViewCell alloc]initWithStyle:UITableViewCellStyleDefault reuseIdentifier:@"CellIdentifier"];
    }
    
    cell.textLabel.text = [[TableViewController getPokedexInfo] objectForKey:[NSNumber numberWithInteger:indexPath.row]];
    
    return cell;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    return [[TableViewController getPokedexInfo] count];
}

+ (NSDictionary *) getPokedexInfo
{
    static NSDictionary * pokedexInfo;
    if (!pokedexInfo)
    {
        pokedexInfo = @{
                         @0 : @"Bulbasaur",
                         @1 : @"Ivysaur",
                         @2 : @"Venosaur",
                         @3 : @"Squirtle",
                         @4 : @"Wartotle",
                         @5 : @"Blastoise",
                         @6 : @"Charmander",
                         @7 : @"Charmeleon",
                         @8 : @"Charizard"
                         };
    }
    return pokedexInfo;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
{
    [TableViewController getOrSetCurrentIndex:[NSNumber numberWithInteger:indexPath.row]];
}

+ (NSNumber *) getOrSetCurrentIndex: (NSNumber *) idx
{
    static NSNumber *index;
    if (!index)
    {
        index = [NSNumber numberWithInteger:0];
    }
    if ([idx integerValue] >= 0)
    {
        index = idx;
    }
    return index;
}

@end

File: Digby/class_10/Pokedex/Pokedex/TableViewController.swift
//
//  TableViewController.swift
//  Pokedex
//
//  Created by Digby Andrews on 2016-07-08.
//  Copyright © 2016 Digby Andrews. All rights reserved.
//

import UIKit

class TableViewController: UITableViewController
{
    internal static var currentIndex = 0
    
    internal static let pokedex =
    [
        ["Name": "Bulbasaur", "Height": "0.7m",],
        ["Name": "Ivysaur", "Height": "1m"],
        ["Name": "Venosaur", "Height": "2m"],
        ["Name": "Squirtle", "Height": "0.4m"],
        ["Name": "Wartotle", "Height": "0.6m"],
        ["Name": "Blastoise", "Height": "2m"],
        ["Name": "Charmander", "Height": "0.9m"],
        ["Name": "Charmeleon", "Height": "1.5m"],
        ["Name": "Charizard", "Height": "3m"]
    ]
    
    override func viewDidLoad()
    {
        super.viewDidLoad()
    }

    override func didReceiveMemoryWarning()
    {
        super.didReceiveMemoryWarning()
    }

    override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int
    {
        return TableViewController.pokedex.count
    }
    
    override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell
    {
        var cell = self.tableView.dequeueReusableCellWithIdentifier("CellId")
        if cell == nil
        {
            cell = UITableViewCell(style: .Default, reuseIdentifier: "CellId")
        }
        
        cell!.textLabel?.text = TableViewController.pokedex [indexPath.row]["Name"]
        return cell!
    }
    
    override func tableView(tableView: UITableView, willSelectRowAtIndexPath indexPath: NSIndexPath) -> NSIndexPath?
    {
        TableViewController.currentIndex = indexPath.row
        return indexPath
    }
    

}
